### PR TITLE
Photocopier Emagging and Toast

### PR DESCRIPTION
**Pull request by TullyBurnalot**
- Hardware upgrade allows photocopiers to be emagged by Cryptographic Sequencers;
- Overloaded laser printing mechanism deals 30 units of Burn Damage to anyone that sits on the photocopier and attempts to copy their posterior. You can now efficiently (?) murder someone by restraining them on a photocopier and printing 6-7 pictures, in sequence;
- Posterior copying left intact. You will still get the picture. Just toasty

:cl:
add: Photocopiers can be emagged
add: Emagged photocopiers deal 30 burn damage to all mobs copying their butts
/:cl:

### CODE REVIEW

File: code/modules/paperwork/photocopier.dm [hunk 1]
```diff
@@ -228,7 +228,18 @@
...
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(ass,"<span class='notice'>Something smells toasty...</span>")
+			var/obj/item/organ/external/G = H.get_organ("groin")
+			G.take_damage(0, 20)
+			sleep(20)
+			ass.emote("scream")
+		if(!ishuman(ass))
```
Comment (TheDZD) [inline code/modules/paperwork/photocopier.dm]: You don't need `if(!ishuman(ass))` here, just have it be `else`.

File: code/modules/paperwork/photocopier.dm [hunk 2]
```diff
@@ -228,7 +228,18 @@
...
 		return
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(ass,"<span class='notice'>Something smells toasty...</span>")
+			var/obj/item/organ/external/G = H.get_organ("groin")
+			G.take_damage(0, 20)
+			sleep(20)
+			ass.emote("scream")
```
Comment (TheDZD) [inline code/modules/paperwork/photocopier.dm]:
```
spawn(20)
	ass.emote("scream")
```
Might be preferable, as it doesn't freeze the entire proc for 2 seconds.

File: code/modules/paperwork/photocopier.dm [hunk 3]
```diff
@@ -228,7 +228,18 @@
...
+		if(!ishuman(ass))
+			var/mob/living/H = ass
```
Comment (TheDZD) [inline code/modules/paperwork/photocopier.dm]: You don't need to cast this, ass is already a `/mob/living`.

File: code/modules/paperwork/photocopier.dm [hunk 4]
```diff
@@ -228,7 +228,17 @@
...
 		return
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(ass,"<span class='notice'>Something smells toasty...</span>")
+			var/obj/item/organ/external/G = H.get_organ("groin")
+			G.take_damage(0, 30)
+			spawn(20)
+				ass.emote("scream")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: H.emote

File: code/modules/paperwork/photocopier.dm [hunk 5]
```diff
@@ -228,7 +228,17 @@
 	var/icon/temp_img
 	if(!check_ass()) //You have to be sitting on the copier and either be a xeno or a human without clothes on.
 		return
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(ass,"<span class='notice'>Something smells toasty...</span>")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: to_chat(H

File: code/modules/paperwork/photocopier.dm [hunk 6]
```diff
@@ -228,7 +228,17 @@
...
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(H,"<span class='notice'>Something smells toasty...</span>")
+			var/obj/item/organ/external/G = H.get_organ("groin")
+			G.take_damage(0, 30)
+			spawn(20)
+				H.emote("scream")
+		else
+			to_chat(ass,"<span class='notice'>Something smells toasty...</span>")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: space between `ass,` and `"`

File: code/modules/paperwork/photocopier.dm [hunk 7]
```diff
@@ -228,7 +228,17 @@
 	var/icon/temp_img
 	if(!check_ass()) //You have to be sitting on the copier and either be a xeno or a human without clothes on.
 		return
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(H,"<span class='notice'>Something smells toasty...</span>")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: space between `H,` and `"`

File: code/modules/paperwork/photocopier.dm [hunk 8]
```diff
@@ -310,6 +320,13 @@
 	else
 		return 1
 
+/obj/machinery/photocopier/emag_act(user as mob)
+	if(!emagged)
+		emagged = 1
+		to_chat(user,"<span class='notice'>You overload the photocopier's laser printing mechanism.</span>")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: you get

File: code/modules/paperwork/photocopier.dm [hunk 9]
```diff
@@ -310,6 +320,13 @@
...
+	else
+		to_chat(user,"<span class='notice'>The photocopier's laser printing mechanism is already overloaded!</span>")
```
Comment (tigercat2000) [inline code/modules/paperwork/photocopier.dm]: the idea

File: code/modules/paperwork/photocopier.dm [hunk 10]
```diff
@@ -228,7 +228,17 @@
 	var/icon/temp_img
 	if(!check_ass()) //You have to be sitting on the copier and either be a xeno or a human without clothes on.
 		return
-
+	if(emagged)
+		if(ishuman(ass))
+			var/mob/living/carbon/human/H = ass
+			to_chat(H, "<span class='notice'>Something smells toasty...</span>")
+			var/obj/item/organ/external/G = H.get_organ("groin")
+			G.take_damage(0, 30)
```
Comment (Fox-McCloud) [inline code/modules/paperwork/photocopier.dm]: why would a laser do brute damage.

Comment (TullyBurnalot) [inline code/modules/paperwork/photocopier.dm]: It doesn't, it does burn.

Comment (Fox-McCloud) [inline code/modules/paperwork/photocopier.dm]: ....*sigh*

We have way too many damage procs.

**Comment by Da-Dman234** [comment]
What happens when the copier runs out of ink, is it still usable to damage people? Also, is there any way to examine said copier to see if it's emagged, or are you gonna have to take a turn on the ass blaster to test if it damages you or not?

**Comment by TullyBurnalot** [comment]
You get burned regardless of ink value. You won't get the ass pic, though.

No special examine text either, time to set your pants on fire to check.

**Pull request merged**